app/models/resource.py

```diff
@@ -44,19 +45,28 @@ def create_resource(name, description, website):
         return result
 
     @staticmethod
-    def generate_fake(count=10):
+    def generate_fake():
+        # TODO: make sure fake resources have users
+
         """Generate count fake Resources for testing."""
         from faker import Faker
 
         fake = Faker()
 
-        # TODO: make sure fake resources have users and addresses
-        for i in range(count):
+        addresses = Address.query.all()
```

Comment:
I'm a big proponent of using [list comprehensions](https://en.wikipedia.org/wiki/Python_syntax_and_semantics#List_comprehensions) instead of for loops when possible. This section from `addresses = Address.query.all()` to the end of the first for loop can be written in one line like this:
```python
unused_addresses = [address for address in Address.query.all() if len(address.resources.all()) == 0]
```
 and if you want to make it even shorter I believe you can do:
```python
unused_addresses = [address for address in Address.query.all() if address.resources.all() == []]
```
It's totally fine if these don't make sense at first, we can go over them step by step at the meeting tomorrow if you'd like (remind me), but I think we should definitely use one of these over the for loop.

*Note: you'll probably have to put a [line break](http://stackoverflow.com/questions/4172448/is-it-possible-to-break-a-long-line-to-multiple-lines-in-python) because the line is too long.

@weronica, what do you think about the readability of the two suggestions I proposed.